test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,169 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+// This file unit tests block manager functions relating to packing and unpacking address cookies.
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <vector>
+
+void
+unpack_addr_cookie_and_check(const uint8_t *packed, uint32_t block_allocsize, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint32_t unpacked_size, unpacked_checksum;
+    uint64_t o, s, c;
+    wt_off_t unpacked_offset;
+    c = o = s = 0;
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &o) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &s) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &c) == 0);
+
+    // Adjust the unpacked values as the block manager also does this to avoid storing large
+    // offsets.
+    unpacked_offset = (s == 0) ? (wt_off_t)o : (wt_off_t)(o + 1) * block_allocsize;
+    unpacked_size = (s == 0) ? (uint32_t)s : (uint32_t)s * block_allocsize;
+    unpacked_checksum = (uint32_t)c;
+
+    if (pack_size != 0) {
+        CHECK(unpacked_offset == pack_offset);
+        CHECK(unpacked_size == pack_size);
+        CHECK(unpacked_checksum == pack_checksum);
+    } else {
+        CHECK(unpacked_offset == 0);
+        CHECK(unpacked_size == 0);
+        CHECK(unpacked_checksum == 0);
+    }
+}
+
+static void
+test_pack_and_unpack_addr_cookie(
+  WT_BLOCK *block, wt_off_t pack_offset, uint32_t pack_size, uint32_t pack_checksum)
+{
+    const uint8_t *begin;
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    uint32_t checksum, obj_id, size;
+    size_t addr_size;
+    wt_off_t offset;
+    pp = p;
+
+    begin = (const uint8_t *)pp;
+    REQUIRE(__wt_block_addr_pack(
+              block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    addr_size = WT_PTRDIFF(pp, begin);
+    unpack_addr_cookie_and_check(begin, block->allocsize, pack_offset, pack_size, pack_checksum);
+
+    // Test the block manager's unpack function.
+    REQUIRE(__wt_block_addr_unpack(
+              NULL, block, begin, addr_size, &obj_id, &offset, &size, &checksum) == 0);
+    if (size != 0) {
```

Comment:
realise that this if statement is the same as `unpack_addr_cookie_and_check`, could we place it into a function as well, then we could call with same validation expectations.

Reply:
They have similar functionalities but are intended to be different as one is the block manager's unpack function while the other is used to test the block manager's pack function.

Reply:
As per discussed lets move the if statement alone to a `check_block` function.